### ISSUE TITLE
Is it important to train with the state dict instead of safetensors?

### ISSUE THREAD

**Issue by w-zhih** [yu-rp/VisualPerceptionToken#9]
Hello, I'm trying to reproduce the training process of Qwen2-VL-2b-VPT-CLIP. But the evaluation results are very low, it seems that it can predict the selected region according to the format, but the value is wrong, and so the final answer is wrong too. I reflect the potential reasons for my failure, and a possible one is that I use the safetensors rather than state dict, as noted in the `Devnote.md`.However, training for one time spends much time, so I want to query that, do you know is this a important thing?

**Comment by yu-rp** [comment]
Hi @w-zhih ,

Thank you for your issue.

Based on our experience, the choice between using state_dict and safetensor does not have a significant impact on training performance. We have been using state_dict mainly because, during the experimental phase—particularly while designing the projector structure—we conducted some analyses on the weights, and found state_dict to be more convenient for such operations. Consequently, we have continued with the state_dict format.

If the training results are unsatisfactory, you might consider diagnosing the issue using the following questions:

1. Have you tried using our evaluation script with the released model? This can help determine whether the issue lies in the evaluation process or the environment configuration.

2. Have you verified that the `safetensor` model was correctly imported? For example, were all parameters properly initialized as expected?

3. Have you monitored the evolution of the training loss? Does it decrease as expected during the training process?

Best,
Runpeng

**Comment by w-zhih** [comment]
Thanks for your advice.
1. I have tried the evaluation script with the released model, and it worked.
2. No problem.
3. The training loss:
![Image](https://github.com/user-attachments/assets/ff278ae3-95d7-4e7b-aae1-fa902c4b06b8)
The eval loss:
![Image](https://github.com/user-attachments/assets/111b72ea-73fd-4e17-bf61-f79e2d93ba4f)
The eval accuracy:
![Image](https://github.com/user-attachments/assets/fdb42fbf-a1d5-4ddb-bb5e-f968f0aa2977)
There seems nothing wrong with these losses.

**Comment by yu-rp** [comment]
Hi @w-zhih ,

Based on the training curves, everything seems to be functioning as expected. In the final figure you provided, the evaluation accuracy appears to be around 50%. In your earlier message, you mentioned concerns about the evaluation performance—may I ask:
-  Is this 50% accuracy falls short of your expectations? 
- Is there a difference between this result and the one produced by the evaluation script

Best,
Runpeng
